### ISSUE TITLE
Feature: Get exact versions from the actual packages installed

### ISSUE THREAD

**Issue by moshest** [lostpebble/generate-package-json-webpack-plugin#11]
Sometimes you want the `package.json` to behave as `package-lock.json` and only have the exact version that actually installed on the project.

I managed to create a small snippet that generate that and i would like to merge it in this package with a non-breaking flag on the options object. 
```ts
const context = module.issuer?.context;
if (!module.external || !context) return;

const moduleName = getNameFromRequest(module.request);

let modulePackageFile;
try {
  modulePackageFile = require.resolve(`${moduleName}/package.json`, {
    paths: [context],
  });
} catch (e) {
  // can't resolve module
  return;
}

let version;
try {
  ({ version } = JSON.parse(fs.readFileSync(modulePackageFile)).toString()));
} catch (e) {
  throw new Error(`Can't parse package.json file: ${modulePackageFile}`);
}

if (!version) {
  throw new Error(`Missing package.json version: ${modulePackageFile}`);
}
```

**Comment by lostpebble** [comment]
Hi @moshest , thanks for opening up this, and those other issues - I'm liking the proposals in all of them!

What your code is doing here is exactly the direction I was thinking of taking in this issue (https://github.com/lostpebble/generate-package-json-webpack-plugin/issues/10#issuecomment-636903301)

I'm happy to merge this in the current library, with an options flag. I was also thinking of doing a major version with a new interface, which simply takes a single options object (allows much more flexibility in future changes and frees us from functionality lock-in on the current first parameter). But perhaps its better not to change it too much yet...

Can experiment with this one, as non-breaking, and then change it after.

As the library gets _slightly_ more complex would be nice to add TypeScript as well. But also, not a rush at the moment.

Would you like to take a crack at adding this snippet? (and perhaps the other suggestions at the same time)? Otherwise I can take a look during the next couple days.

**Comment by lostpebble** [comment]
Thanks so much @moshest ! The changes look great. Sorry for late merge, I was out for the afternoon.

Just deployed `1.1.0` - hopefully should all be working smoothly.

**Comment by lostpebble** [comment]
Oops seems like I've run into an issue. Didn't test it all properly. Will need to quickly fix and deploy a patch version.

` TypeError: Cannot destructure property 'additionalDependencies' of '(intermediate value)(intermediate value)(intermediate value)' as it is undefined.`

---

Just deployed `1.1.1` with the fix.

**Comment by moshest** [comment]
Pushed another fix here #17
